### PR TITLE
[WFLY-11145]: multicastPrefix and anycastPrefix cannot be set

### PR DESCRIPTION
**Pull request by ehsavoie**
Adding extra parameters on connectors and acceptors when constructing
TransportConfigurations.

Jira: https://issues.jboss.org/browse/WFLY-11145

### CODE REVIEW

File: messaging-activemq/pom.xml [hunk 1]
```diff
@@ -244,5 +244,10 @@
             <scope>test</scope>
             <classifier>tests</classifier>
         </dependency>
+        <dependency>
+            <groupId>org.apache.activemq</groupId>
+            <artifactId>artemis-stomp-protocol</artifactId>
+            <version>2.6.3.jbossorg-001</version>
```
Comment (jbliznak) [inline messaging-activemq/pom.xml]: unparametrized version does not seem correct

Comment (ehsavoie) [reply messaging-activemq/pom.xml]: Sorry this was a leftover

**Review by jmesnil — CHANGES_REQUESTED** [review]
PR looks good except it missing conversion of the extra parameters name.

WildFly convention is to use lower-case-with-dash for its parameters (e.g. `multicast-prefix`) while Artemis uses CamelCases (e.g. `multicastPrefix`).

The `TransportConfigOperationHandlers` is holding the mapping of the parameters to convert from WildFly convention to Artemis.

This means the test must also be changed so that it is creating the resource with:

```
parameters.put("multicast-prefix", "jms.topic");
             parameters.put("anycast-prefix", "jms.queue");
             ops.addHttpConnector("http-test-connector", "test-binding", "http-acceptor", parameters);
```

**Comment by ehsavoie** [comment]
I don't fully agree with you @jmesnil : we don't support those parameters as parameters but as extra parameters and as such we currently don't "list" them, they could be anything as there is no defined list (like for connectors / acceptors) from Artemis.

**Comment by jmesnil** [comment]
@ehsavoie fair enough, these extraParameters are "get-out-of-jail" parameters that could be anything...

However from WildFly perspective, they are passed to the `params` attribute of the `connector` so we end up with config using 2 different naming conventions:

```
 <http-connector name="http-connector-throughput"
                            socket-binding="http"
                            endpoint="http-acceptor-throughput">
   <param name="batch-delay" value="50"/>
   <param name="multicastPrefix" value="jms.topic"/>
</http-connector>
```

This is inconsistent and will be difficult to document and maintain.

Maybe we could offer conversion for some well-known and documented parameters such as `multicast-prefix`and `unicast-prefix`.
If we can not find such a mapping, so be it, we pass the names as they are to Artemis.

wdyt?

**Comment by ehsavoie** [comment]
retest this please

**Comment by jmesnil** [comment]
retest this please

**Review by jmesnil — APPROVED** [review]

**Pull request merged**